### DISCUSSION TITLE
Do redis-cluster scalers work with Azure Redis

### DISCUSSION THREAD

**Discussion by pinkfloydx33** [kedacore/keda#3306] [Q&A / Need Help] [answered]
Do the redis-cluster scalers work with Azure Redis in cluster mode? 

When configured as a cluster, Azure still only provides a single host name and does not expose the individual node addresses. The expectation is that the client will follow the `MOVE` command; Azure instructions explicitly call out that the host address is the same one you'd use in non-cluster mode.

So I'm wondering whether the redis-cluster scalers will work with Azure Redis in cluster mode if all we're able to do is provide a single host name. This is not clear from the documentation or the samples which imply that we should be passing multiple addresses.

**Comment by tomkerkhove** [comment]
As far as I know it should work, but if it doesn't then we have to get that fixed.

**Comment by pinkfloydx33** [comment, marked as answer]
We're currently in the process of planning out some infrastructure needs for a greenfield project. We are considering Azure redis cluster, but as we already know that we will need keda scaler support we're afraid to make this decision without confirmation that the *single* host address will work. 

Azure doesn't allow you to "un-cluster" an instance once you've enabled clustering and testing it myself would require setting up a ton of infrastructure just to verify. Hence my question.

If someone has a definitive answer it'd be much appreciated (and perhaps the docs could be made clearer in either case). TIA

**Reply by tomkerkhove** [reply]
I've opened https://github.com/kedacore/keda/issues/3325 to automaticall test this but we'll need somebody to pick this up first. If you are interested, I'm happy to spin up an instance for this.

**Reply by JorTurFer** [reply]
Maybe you can create the redis with the C0 tier now, I think that test it (and fix it in case of problems) is a good addition for next release, I'll tackle it next week if there isn't any volunteer

**Reply by pinkfloydx33** [reply]
Clustering is only available on the premium and enterprise tiers. You'd need `P1` at a minimum

**Reply by JorTurFer** [reply]
I have tried deploying a P1 instance and I have been able to use [`redis-list`](https://keda.sh/docs/2.7/scalers/redis-lists/) scaler when the redis cluster doesn't have the clustering option enabled.
![image](https://user-images.githubusercontent.com/36899226/177630285-3c355a02-b31a-4e51-bdcc-e4718f00a141.png)
Now I'm changing the cluster type (enabling the clustering) and I will try it again

**Reply by JorTurFer** [reply]
I have checked again, and I have been able to use [`redis-cluster`](https://keda.sh/docs/2.7/scalers/redis-cluster-lists/) scaler with Azure Redis Cluster once I enabled the clustering and set 2 instances (in cluster mode with only one didn't work)
![image](https://user-images.githubusercontent.com/36899226/177647403-6e46bc37-0e01-4c0c-9b26-19f8b3a47f41.png)

**Reply by pinkfloydx33** [reply]
So redis-cluster successfully worked with the single host address is what you are saying?

**Reply by JorTurFer** [reply]
yes, this is my config:
```yaml
  triggers:
    - authenticationRef:
        name: azure
      metadata:
        enableTLS: 'true'
        listLength: '5'
        listName: test
      type: redis-cluster
```
Adding by trigger the addresses and password. The `addresses` value is `keda.redis.cache.windows.net:6380`

**Reply by pinkfloydx33** [reply]
Thank you for verifying this case for us! Much appreciated

**Reply by JorTurFer** [reply]
happy to help :)

**Comment by JorTurFer** [comment]
do we still need e2e test to cover this case? Azure Redis Cache with premium tier is expensive
WDYT @tomkerkhove ?

**Reply by pinkfloydx33** [reply]
Yeah premium tier is expensive.... Yall don't have a free dev/test subscription provided by Msft or anything like that?

**Reply by JorTurFer** [reply]
yes we have, but having Azure Redis Cluster with 2 shards means > 600€/month (needing >30 min to spin up the cluster, I think that we cannot create it on-demand) and I'm not sure that it gives any extra value. It works, I have tested it today, and we already have e2e test to cover those scalers with selfhosted redis cluster, using 1 or more domains in the `addresses` is something that could be tested without using azure services. Why not using elastic cache from AWS? or the google service for it? 
I mean, do we really need to test scalers in Azure/AWS/GCP if we can test it with self-hosted services (and using them in cloud provider doesn't give any extra value like authentications or similar)?´
Thoughts @kedacore/keda-maintainers ?

**Reply by pinkfloydx33** [reply]
Oh, I'm not suggesting that you go either way (self hosted makes more sense imo, now that you've verified explicitly). I was just surprised they don't give you a free subscription, especially given they embed keda in the azure functions cli. /shrug

**Reply by JorTurFer** [reply]
We have a subscription on the bigget cloud providers, but the credit is not infinite xD

**Reply by JorTurFer** [reply]
Apart from that, it's not only the credit, it's maintaining a test case that it's not testing anything new (at least IMO)

**Reply by tomkerkhove** [reply]
Let's move the conversation to https://github.com/kedacore/keda/issues/3325
